x/tx/signing/textual/any.go

```diff
@@ -49,10 +49,15 @@ func (ar anyValueRenderer) Format(ctx context.Context, v protoreflect.Value) ([]
 		return nil, err
 	}
 
-	// The Any value renderer suppresses emission of the object header
+	// The Any value renderer suppresses emission of the object header for all
+	// messages that go through the messageValueRenderer.
 	omitHeader := 0
 	_, isMsgRenderer := vr.(*messageValueRenderer)
-	if isMsgRenderer && subscreens[0].Content == fmt.Sprintf("%s object", internalMsg.ProtoReflect().Descriptor().Name()) {
+	if isMsgRenderer {
+		if subscreens[0].Content != fmt.Sprintf("%s object", internalMsg.ProtoReflect().Descriptor().Name()) {
+			return nil, fmt.Errorf("any internal message expects %s, got %s", fmt.Sprintf("%s object", internalMsg.ProtoReflect().Descriptor().Name()), subscreens[0].Content)
```

Comment:
If the first condition `isMsgRenderer` is true, then the second condition also must be true, or else it's an implementation error upstream (e.g. in the protobuf values passed into Textual), and the user should be aware of it.

I propose to return an error here.

Reply:
@JimLarson I'll merge this, we can re-discuss this in my follow up pr